### ISSUE TITLE
Potential bug in the MAP@R metric

### ISSUE THREAD

**Issue by amirassov** [jzakirov/huawei2020#3]
Hello! 

I tried to adapt `MAP@R` metric for my task and found out that your implementation has a bug. I just took examples from [the original paper](https://arxiv.org/abs/2003.08505) and put into your function:
<img width="727" alt="Screenshot 2021-09-20 at 12 00 53" src="https://user-images.githubusercontent.com/14277430/133987208-93c5f626-6ab5-4c25-982d-61be0fb88509.png">

```python
import torch
import numpy as np


conformity_matrix = torch.tensor([[True for _ in range(10)] + [False for _ in range(10)]])

permutation_matrix = torch.tensor([[0, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 1, 2, 3, 4, 5, 6, 7, 8, 9]])
np.abs(map_at_k(permutation_matrix, conformity_matrix, topk=None) - 0.1) <= 1e-6
# False

permutation_matrix = torch.tensor([[0, 10, 11, 12, 13, 14, 15, 16, 17, 1, 18, 19, 2, 3, 4, 5, 6, 7, 8, 9]])
np.abs(map_at_k(permutation_matrix, conformity_matrix, topk=None) - 0.12) <= 1e-6
# False

permutation_matrix = torch.tensor([[0, 1, 11, 12, 13, 14, 15, 16, 17, 18, 19, 10, 2, 3, 4, 5, 6, 7, 8, 9]])
np.abs(map_at_k(permutation_matrix, conformity_matrix, topk=None) - 0.2) <= 1e-6
# False

permutation_matrix = torch.arange(20).reshape(1, 20)
np.abs(map_at_k(permutation_matrix, conformity_matrix, topk=None) - 1.0) <= 1e-6
# False
```

All tests pass correctly if we change line https://github.com/zakajd/huawei2020/blob/master/src/callbacks.py#L91 by: 
```python 
average_precision = precision.sum(dim=-1) / R
```

**Comment by jzakirov** [comment]
Hi, 
Thanks for noticing! Fixed
